ruuvi_examples/ruuvi_firmware/main.c

```diff
@@ -282,97 +296,149 @@ ret_code_t lis2dh12_int2_handler(const ruuvi_standard_message_t message)
+                        nrf_delay_ms(1000u);  }
+ 
+
+  if (model_plus) GREEN_LED_ON;    // will be turned off in power_manage.
+
+  nrf_delay_ms(MAIN_LOOP_INTERVAL_RAW + 100); // Delay before advertising so we get valid data on first packet
+
+  bluetooth_advertising_start(); NRF_LOG_INFO("Advertising start  \r\n");           // can this fail ?
+
+  if (init_status) {     // Report on status of initalization 
+   url_buffer[9]=init_status>>8;url_buffer[10]=init_status & 0xFF;       //  URL ruu.vi/#xx
+   bluetooth_set_eddystone_url(url_buffer, sizeof(url_buffer) ); nrf_delay_ms(500u); // keep sending this for a while
+//  raspberry pi:  
+//  export bdaddr='zz yy xx ww vv uu' #  MAC address least significant byte FIRST
+//  hcidump --raw |g  --line-buffered --after-context=2 $bdaddr
+//  04 3E 2B 02 01 00 01 C2 24 68 3C 10 C7 1F 02 01 06 03 03 AA
+//                       zz yy xx ww vv uu
+//  FE 17 16 AA FE 10 F6 03 72 75 75 2E 76 69 2F 23 10 00 00 00
+//                          r  u  u  .  v  i  /  #  ^^^^^--init_status example BATTERY_FAILED_INIT
+
+  // like format 03  too
+   data_buffer[0]   =   0x13 ;                              // NOT SENSOR_TAG_DATA_FORMAT 03;  (but could be intreperted as such)
+   data_buffer[1]   =    254;                                                          // NOT data->humidity;
+   data_buffer[2]   =    125;                data_buffer[3]  =    125;                 // NOT data->temperature   
+   data_buffer[4]   =  11125        >>8;     data_buffer[5]  =   11125       & 0xFF;   // NOT data->pressure       
+   data_buffer[6]   =  0x1D1D       >>8;     data_buffer[7]  =   0x1D1D      & 0xFF;   // NOT accx              I did 
+   data_buffer[8]   =  0x0BAD       >>8;     data_buffer[9]  =   0x0BAD      & 0xFF;   // NOT accy                     bad
+   data_buffer[10]  = init_status   >>8;     data_buffer[11] =  init_status  & 0xFF;   // NOT accz                xx xx
+   data_buffer[12]  =    vbat       >>8;     data_buffer[13] =    vbat       & 0xFF;
+//  from raspberry pi  
+//      export bdaddr='zz yy xx ww vv uu' MAC address least Significant byte FIRST
+//      hcidump --raw |  grep --line-buffered --after-context=2  "$bdaddr"
+//      > 04 3E 2B 02 01 00 01 C2 24 68 3C 10 C7 1F 02 01 06 1B FF 99
+//                             zz yy xx ww vv uu
+//        04 13 FE 7D 7D 2B 75 1D 1D 0B AD 10 00 08 41 00 00 00 00 00
+//                        +  u I did   bad sssss vv vv   ssss is init status example BATTERY_FAILED_INIT; vv vv=vbat=0x0841 ie 2.113v
+  bluetooth_set_manufacturer_data(data_buffer, sizeof(data_buffer)); nrf_delay_ms(500u); // keep sending this for a while
+                     }            
+
+  // Enter main loop. spinning here interrupts will bring us out frequently
+  // accelerometer invokes lis2dh12_int2_handler
   for (;;)
   {
     app_sched_execute();
-    power_manage();
+    power_manage();         // calls sd_app_evt_wait
   }
 }
+
+
+
+
+/* vim: set columns=140: */

```

Comment:
Clarification of the comments is always good. 

In general any change that is seen by the factory tester or end-users requires a careful cost-benefit analysis. Changing anything - even fixing bugs - breaks someone's project.

Battery voltage check is great for the factory and first time boot, and since the program does not halt on error it does not prevent old tags from booting up. I'm somewhat concerned about the factory process: If a tag which has excessive voltage drop (or any other issue) on the tag can boot, some faulty tags could pass the QC process. I think proper way of addressing faults would be to use the data format 5 "invalid" values, as these would be seen in every transmission

Reply:
I tried not to change anything that would effect transmitted data (other that the initial additional error packets when there is a problem). I don't see anything that would break any existing receiving program, that's why I made the error packet similar to format 03 (with illegal values) but with a different format code.

Thank you for you comments. I will make some changes and let you know.

PS I will make up a video of successful and failing tags so you can see "how nice " it looks.